datafusion/optimizer/src/optimizer.rs

```diff
@@ -274,6 +284,77 @@ impl Optimizer {
         debug!("Optimizer took {} ms", start_time.elapsed().as_millis());
         Ok(new_plan)
     }
+
+    fn optimize_node(
```

Comment:
What value does this function add? In other words, why not call `rule.try_optimize` directly at the callsite 🤔

Reply:
I forgot to add future TODO:
we can do batch optimize
```rust
for rule in rules:
    rule.try_optimize(rule)
```